flycheck.el

```diff
@@ -6924,7 +6924,16 @@ See URL `http://www.erlang.org/'."
 
 (defun flycheck-rebar3-project-root (&optional _checker)
   "Return directory where =rebar.config= is located."
-  (locate-dominating-file buffer-file-name "rebar.config"))
+  (let* ((rebarconfig (locate-dominating-file buffer-file-name "rebar.config"))
+         (project-dir (if rebarconfig
+                          (file-name-as-directory
+                           (substring rebarconfig
+                                      0
+                                      (string-match
+                                       "\\(/_build/[^/]+/lib\\|/_checkouts\\)"
+                                       rebarconfig)))
```

Comment:
Chances are, if flycheck encounters this then it's a mistake on the part of the user. Something got built in a way it wasn't supposed to, or some artifact of some build survived somewhere it shouldn't have. No matter the case, the top `rebar.config` it can find is always the right answer. I think something less eager would still work 90% of the time, so if we're concerned about speed we could opt for something like that.

There's no nesting of projects with rebar. If you are doing something like, there's a way to structure your source so it will build multiple applications, but if you did that, it would still be done with a single `rebar.config` file and the applications would be bundled together into a single release artifact. 

The closest thing to what I think you're looking for would be if you had two rebar3 projects, and a third than included both of them as dependencies, but that's already covered.

Did that clear it up?